src/HexViewer.ts

```diff
@@ -291,16 +303,49 @@ export class HexViewer {
     }
 
     public deselect() {
-        this.setSelection(-1, -1);
+        this.setCursor(-1);
     }
 
-    public setSelection(start: number, end?: number) {
+    public setCursor (cursor: number) {
+        this.selectionCursor = cursor;
+        this.setSelection(
+            this.selectionCursor
+        );
+    }
+
+    public shiftCursor (cursor: number) {
+        var start = this.selectionStart;
+        var end = this.selectionEnd;
+        if (this.selectionCursor === end) {
+            end = cursor;
+        } else if (this.selectionCursor === start) {
+            start = cursor;
+        }
+        this.selectionCursor = cursor;
+        this.setSelection(
+            Math.max(0, start),
+            end
+        );
+    }
```

Comment:
Hopefully these methods won't be needed at all when you eliminate the `selectionCursor` variable (https://github.com/kaitai-io/kaitai_struct_webide/pull/119#discussion_r460407701).